src/main/java/com/github/fppt/jedismock/operations/lists/ListPopper.java

```diff
@@ -11,28 +11,26 @@
 import java.util.List;
 
 abstract class ListPopper extends AbstractRedisOperation {
-    private final Slice key;
-    final List<Slice> list;
 
     ListPopper(RedisBase base, List<Slice> params) {
         super(base, params);
-        key = params.get(0);
-        RMList listDBObj = getListFromBaseOrCreateEmpty(key);
-        list = listDBObj.getStoredData();
     }
 
     abstract Slice popper(List<Slice> list);
 
-    private Slice pop() {
+    private Slice pop(Slice key, List<Slice> list) {
         Slice result = popper(list);
+        base().markKeyModified(key);
         if (list.isEmpty()) {
             base().deleteValue(key);
         }
         return result;
     }
 
     protected final Slice response() {
-
+        Slice key = params().get(0);

```

Comment:
We can't really validate parameters in constructor, because exception will occur in reflection and it is rather hard to get correct exception there